src/compiler/parser.ts

```diff
@@ -2286,27 +2311,31 @@ namespace ts {
             returnToken: SyntaxKind.ColonToken | SyntaxKind.EqualsGreaterThanToken,
             flags: SignatureFlags,
             signature: SignatureDeclaration): void {
-            if (!(flags & SignatureFlags.JSDoc)) {
-                signature.typeParameters = parseTypeParameters();
-            }
-            signature.parameters = parseParameterList(flags);
+            doOutsideOfExpressionContext(fillSignatureWorker);

```

Comment:
Because of how complicated arrow function speculative parsing gets, it is much easier to manage the context within `fillSignature` than at the many speculative branches for arrow function parsing.